MultiThreadDemoDlg.cpp

```diff
@@ -0,0 +1,438 @@
+	commthread->minusevent = CreateEvent(NULL,TRUE,FALSE,NULL);
+	
+	return TRUE;  // return TRUE  unless you set the focus to a control
+}
+
+void CMultiThreadDemoDlg::OnSysCommand(UINT nID, LPARAM lParam)
+{
+	if ((nID & 0xFFF0) == IDM_ABOUTBOX)
+	{
+		CAboutDlg dlgAbout;
+		dlgAbout.DoModal();
+	}
+	else
+	{
+		CDialogEx::OnSysCommand(nID, lParam);
+	}
+}
+
+// If you add a minimize button to your dialog, you will need the code below
+//  to draw the icon.  For MFC applications using the document/view model,
+//  this is automatically done for you by the framework.
+
+void CMultiThreadDemoDlg::OnPaint()
+{
+	if (IsIconic())
+	{
+		CPaintDC dc(this); // device context for painting
+
+		SendMessage(WM_ICONERASEBKGND, reinterpret_cast<WPARAM>(dc.GetSafeHdc()), 0);
+
+		// Center icon in client rectangle
+		int cxIcon = GetSystemMetrics(SM_CXICON);
+		int cyIcon = GetSystemMetrics(SM_CYICON);
+		CRect rect;
+		GetClientRect(&rect);
+		int x = (rect.Width() - cxIcon + 1) / 2;
+		int y = (rect.Height() - cyIcon + 1) / 2;
+
+		// Draw the icon
+		dc.DrawIcon(x, y, m_hIcon);
+	}
+	else
+	{
+		CDialogEx::OnPaint();
+	}
+}
+
+// The system calls this function to obtain the cursor to display while the user drags
+//  the minimized window.
+HCURSOR CMultiThreadDemoDlg::OnQueryDragIcon()
+{
+	return static_cast<HCURSOR>(m_hIcon);
+}
+
+
+void CMultiThreadDemoDlg::OnTimer(UINT_PTR nIDEvent)
+{
+	
+	strMain.Format("Running: %d", currValue);
+	if(currValue <= maxValue)
+	{
+		if(currValue == maxValue)
+		{
+			currValue = 0;
+		}
+		else
+		{
+			currValue++;
+		}
+			
+	}
+
+	m_timer.SetWindowText(strMain);
+
+	timethread -> strMain.Format("Running: %d", timethread -> currentValue);
+	m_thread.SetWindowText(timethread ->strMain);
+	CDialogEx::OnTimer(nIDEvent);
+}
+
+
+UINT TimerThread::MyThreadProc(LPVOID Param)
+{	
+	
+	TimerThread *timethreada =    (TimerThread*)(Param);
+
+	/* check the other event state and decide what to do */	
+	if( WaitForSingleObject(timethread -> stopevent,10) == WAIT_OBJECT_0)	 
+	{
+
+		timethreada->currentValue = 0;
+
+		while(timethreada -> currentValue <= timethreada -> maxValue +1 )
+		{
+			if(timethreada -> currentValue == timethreada -> maxValue +1 )
+			{
+				timethreada -> currentValue = 0;
+			}
+
+			Sleep(1050);
+			timethreada -> currentValue++;
+			if ( WaitForSingleObject(timethread -> stopevent,5) == WAIT_TIMEOUT )
+			{
+				TRACE("The No butoon still run \n");
+
+				// Don't forget to the release the event when finished the events!
+				// Release the source, if don't do this step, the "Deathlock" will be appeared after use the thread again.
+				CloseHandle(timethread->stopevent);
+
+				return TRUE;
+			}
+				
+		}
+	}
+	
+}
+
+UINT CommThread::AddThreadProc(LPVOID Param)
+{
+	/* check the other event state and decide what to do */
+	SetEvent(commthread->addevent);
+
+	if(WaitForSingleObject(commthread -> minusevent, 0) != WAIT_OBJECT_0 || WaitForSingleObject(commthread -> minusevent, 0) == WAIT_OBJECT_0 )

```

Comment:
The meaning is to consider the first trigger without any status from othre threads (Not equal to zero)
Another one is that the the trigger after first one and other threads have benn triggered (Equal to zero)

Reply:
can it change to this:

if((WaitForSingleObject(commthread -> minusevent, 0) != WAIT_OBJECT_0 )
{}
else // consider the situation besides the nosignaled state
{}

and other how to consider the situation the first trigger